hs_core/hydroshare/resource.py

```diff
@@ -876,6 +876,58 @@ def delete_resource(pk):
     return pk
 
 
+def get_resource_file_name(f):
+    '''
+    get the file name of a specific ResourceFile object f
+    Args:
+        f: the ResourceFile object to return name for
+    Returns:
+        the file name of the ResourceFile object f
+    '''
+    if f.resource_file:
+        # file was originally from local disk, and is currently stored in irods hydroshare zone
+        res_fname = f.resource_file.name
+    elif f.fed_resource_file:
+        # file was originally from local disk, but is currently stored in federated irods zone
+        res_fname = f.fed_resource_file.name
+    else:
+        # file was originally from federated irods zone, and is currently stored in the
+        # same federated irods zone
+        res_fname = f.fed_resource_file_name_or_path
+    return res_fname
+
+
+def delete_resource_file_only(resource, f):
```

Comment:
@pkdash I initially planned to add this as a class method in ResourceFile class, but then I found this ```delete_resource_file()``` method in resource.py, which emits signals and deletes relevant metadata elements in addition to deleting the resource file, so I decided to put this method ```delete_resource_file_only()``` right before the existing ```delete_resource_file()``` method which calls ```delete_resource_file_only()``` for file deletion. Perhaps it is preferrable to move this  method to class method in ResourceFile class, but leaving the existing ```delete_resource_file()``` here as a utility function, or perhaps it is better just to move both into class methods in ResourceFile class, or just leave them both here. I'll defer to you on this then.